### ISSUE TITLE
Make Context's fields public or have a fn new?

### ISSUE THREAD

**Issue by kamiyaa** [kkawakam/rustyline#211]
There doesn't seem to be a way to initialize a `Context` so it can be used with `completer.update()`?
Or maybe I'm missing something here

**Comment by gwenn** [comment]
There is no `ctx: &Context` param in [update](https://docs.rs/rustyline/*/rustyline/completion/trait.Completer.html#method.update) method signature.
You need access to history in your `update` implementation ?

**Comment by kamiyaa** [comment]
My bad, I meant completer.complete()

**Comment by gwenn** [comment]
Oh, I see:
https://github.com/kamiyaa/joshuto/blob/c3919c91a866a7fd520d7dcb8b99c7d4aafbbb93/src/textfield.rs#L119
```rust
                            let res = completer.complete(line_buffer.as_str(), line_buffer.len());
```
`complete` is invoked manually...
I am not sure what is the right solution:
- introduce a public `complete_file` method to `FilenameCompleter`
- introduce a new `complete` method to `Completer` trait. And rename the current one `complete_with_context`
- ...
?

**Comment by kamiyaa** [comment]
I was thinking of just (for my use case at least) making the fields of `Context` public, or having a way to instantiate a dummy `Context`, that I pass into `complete()` that does basically nothing (works like old `complete`). But I'm not really sure what the new `complete()` does with the new context parameter.

**Comment by gwenn** [comment]
Would you mind giving a try to https://github.com/kkawakam/rustyline/pull/212 ?

**Comment by kamiyaa** [comment]
Looks good

**Comment by gwenn** [comment]
Should be fixed by release 4.1
